crates/runtime/src/search/util.rs

```diff
@@ -14,44 +14,66 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 #![allow(clippy::implicit_hasher)]
+
 use std::{collections::HashMap, sync::Arc};
 
 use app::App;
 use datafusion::{common::Constraint, datasource::TableProvider, sql::TableReference};
 use datafusion_federation::FederatedTableProviderAdaptor;
+use search::generation::CandidateGeneration;
 use snafu::ResultExt;
 use tokio::sync::RwLock;
 
 use crate::accelerated_table::AcceleratedTable;
 use crate::datafusion::{DataFusion, SPICE_DEFAULT_CATALOG, SPICE_DEFAULT_SCHEMA};
 
 use crate::embeddings::table::EmbeddingTable;
+use crate::search::SearchGenerationSnafu;
+
+use super::{Error, Result, full_text::table::TableWithFullText};
 
-/// If a [`TableProvider`] is an [`EmbeddingTable`], return the [`EmbeddingTable`].
-/// This includes if the [`TableProvider`] is an [`AcceleratedTable`] with a [`EmbeddingTable`] underneath.
-pub(super) async fn get_embedding_table(
```

Comment:
A bad diff. This change abstracts `get_embedding_table` (which returns a specific table provider `EmbeddingTable`), to check for an arbitrary concrete table provider.

Reply:
e.g.
```rust
find_concrete_table_provider::<EmbeddingTable>(&table_provider)
```